### PR TITLE
Refactor FieldValueConverterRegistryPass not to use static callbacks

### PR DESCRIPTION
**Pull request by alongosz**
This PR refactors `eZ\Publish\Core\Base\Container\Compiler\Storage\Legacy\FieldValueConverterRegistryPass` not to use static callback when registering concrete `FieldValueConverter`s.

This change simplifies injecting any dependency into any `FieldValueConverter` (and makes the code cleaner).

*On the other hand, to compensate for "losing" laziness on Converters, Registry is now lazy.*

**TODO**:
- [x] Remove static callback logic from the `FieldValueConverterRegistryPass`.
- [x] Deprecate static callback method `create` in concrete `FieldValueConverter` implementations.
- [x] Make `ezpublish.persistence.legacy.field_value_converter.registry` lazy
- [x] Remove `lazy` and `callback` attributes from `ezpublish.storageEngine.legacy.converter` tagged-services.
- [x] Align unit tests.
- [x] Remove TMP commit & rebase after #1865 is merged - for Travis to work again.

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/AuthorConverter.php [hunk 1]
```diff
@@ -18,18 +18,6 @@
 class AuthorConverter implements Converter
 {
     /**
-     * Factory for current class.
-     *
-     * Note: Class should instead be configured as service if it gains dependencies.
-     *
-     * @return Author
-     */
-    public static function create()
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/AuthorConverter.php]: you should deprecate and not remove directly

**Comment by alongosz** [comment]
Restored `::create` and deprecated it in 99903c2.

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/FloatConverter.php [hunk 2]
```diff
@@ -22,11 +22,13 @@ class FloatConverter implements Converter
     const HAS_MAX_VALUE = 2;
 
     /**
+     * @deprecated since 6.8, will be removed in 7.x, use default constructor instead.
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/FloatConverter.php]: nitpic, but `@deprecated` is no the title of the doc block :)

so place it after "NOTE:" part.

Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/FloatConverter.php]: Ok, fixed in eb1b693 :)

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ISBNConverter.php [hunk 3]
```diff
@@ -5,6 +5,8 @@
  *
  * @copyright Copyright (C) eZ Systems AS. All rights reserved.
  * @license For full copyright and license information view LICENSE file distributed with this source code.
+ *
+ * @version
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ISBNConverter.php]: missed this one now

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/FloatConverter.php [hunk 4]
```diff
@@ -22,13 +22,13 @@ class FloatConverter implements Converter
...
-     * @deprecated since 6.8, will be removed in 7.x, use default constructor instead.
-     *
      * Factory for current class.
      *
      * Note: Class should instead be configured as service if it gains dependencies.
      *
-     * @return float
+     * @deprecated since 6.8, will be removed in 7.x, use default constructor instead.
+     *
+     * @return Float
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/FloatConverter.php]: and this

**Comment by alongosz** [comment]
Added missing CS fixes in affa731 (sorry!).

**Comment by pspanja** [comment]
From comment at https://github.com/ezsystems/ezpublish-kernel/pull/1812, should this be closed now?

**Comment by andrerom** [comment]
> From comment at #1812, should this be closed now?

Can, but this is a nice cleanup and already done, so would like to merge it for 6.8

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/FloatConverter.php [hunk 5]
```diff
@@ -26,7 +26,9 @@ class FloatConverter implements Converter
      *
      * Note: Class should instead be configured as service if it gains dependencies.
      *
-     * @return Float
+     * @deprecated since 6.8, will be removed in 7.x, use default constructor instead.
+     *
+     * @return float
```
Comment (pspanja) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/FloatConverter.php]: Nitpick: `FloatConverter` :)

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/IntegerConverter.php [hunk 6]
```diff
@@ -26,7 +26,9 @@ class IntegerConverter implements Converter
      *
      * Note: Class should instead be configured as service if it gains dependencies.
      *
-     * @return Integer
+     * @deprecated since 6.8, will be removed in 7.x, use default constructor instead.
+     *
+     * @return int
```
Comment (pspanja) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/IntegerConverter.php]: `IntegerConverter`

**Comment by alongosz** [comment]
Looks like almost every converter has wrong doc on `create` method return type. Since we're doing a cleanup I updated all of them in 548fede (using FQCN as we do everywhere).

**Comment by alongosz** [comment]
It's ready now.
I had to rebase this to be able to run composer on Travis again.
Since review is done I also squashed all commits.

// cc @andrerom

**Comment by andrerom** [comment]
Thanks @alongosz

**Pull request merged**